### PR TITLE
Microshift command docs

### PR DESCRIPTION
**Pull request by soltysh**
/assign @ardaguclu @ShaunaDiaz

### CODE REVIEW

File: tools/gendocs/gen_openshift_docs.go [hunk 1]
```diff
@@ -51,12 +53,16 @@ func main() {
 	out := os.Stdout
 	cmd := cli.NewOcCommand(&bytes.Buffer{}, out, ioutil.Discard)
 
-	outFile := filepath.Join(outDir, "oc-by-example-content.adoc")
+	fileName := "oc-by-example-content.adoc"
```
Comment (ShaunaDiaz) [inline tools/gendocs/gen_openshift_docs.go:56]: Should this be "microshift-oc-by-example.adoc"?

File: tools/gendocs/gendocs/microshift.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package gendocs
+
+import (
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// a set of commands excluded from microshift
```
Comment (ShaunaDiaz) [inline tools/gendocs/gendocs/microshift.go:7]: Are excluding these, or adding just the ones in the Google doc?

**Review by ShaunaDiaz — CHANGES_REQUESTED** [review]

File: tools/gendocs/gen_openshift_docs.go (hunk 1 above)
Comment (dhellmann) [reply tools/gendocs/gen_openshift_docs.go:56]: The prefix is added down on lines 60-62, so I think this is OK as a base filename.

File: tools/gendocs/gendocs/gendocs.go [hunk 3]
```diff
@@ -70,18 +70,21 @@ func GenDocs(cmd *cobra.Command, filename string, admin bool) error {
...
 }
 
-func extractExamples(cmd *cobra.Command, admin bool) Examples {
+func extractExamples(cmd *cobra.Command, admin, microshift bool) Examples {
 	objs := Examples{}
 	for _, c := range cmd.Commands() {
 		if len(c.Deprecated) > 0 {
 			continue
 		}
+		if microshift && microshiftCommands.Has(c.CommandPath()) {
```
Comment (dhellmann) [inline tools/gendocs/gendocs/gendocs.go:79]: nit: It's a little odd to call it `microshiftCommands` when it is the list of exclusions.

File: tools/gendocs/gendocs/microshift.go (hunk 2 above)
Comment (dhellmann) [reply tools/gendocs/gendocs/microshift.go:7]: The logic in gendocs.go on lines 79-81 says that if it sees a command in this list it skips producing output.

**Review by dhellmann — COMMENTED** [review]
The list of commands looks good to me. I had one nit about the variable name, but it's not worth holding things up. I'll leave a hold in case you want other contributors to look at it before merging, but you can remove that if you're ready to merge it.

/lgtm
/hold

**Comment by soltysh** [comment]
The problem in e2e-aws-ovn is not related with this PR, which is only modifying the docs generation helper, but it's being tracked separately in https://issues.redhat.com/browse/OCPBUGS-8092

/override ci/prow/e2e-aws-ovn

**Pull request merged**